scripts/operations/xnamevalidation.sh

```diff
@@ -0,0 +1,438 @@
+	yq w -i "${TMPDIR}/customizations.yaml" -- 'spec.kubernetes.services.cray-opa.opa.xnamePolicy.ckdump' 'true'
+	yq w -i "${TMPDIR}/customizations.yaml" -- 'spec.kubernetes.services.cray-opa.opa.xnamePolicy.dvs' 'true'
+	yq w -i "${TMPDIR}/customizations.yaml" -- 'spec.kubernetes.services.cray-opa.opa.xnamePolicy.enabled' 'true'
+	yq w -i "${TMPDIR}/customizations.yaml" -- 'spec.kubernetes.services.spire.server.tokenService.enableXNameWorkloads' 'true'
+}
+
+# disable_xnameValidation_in_charts uses yq to disable xname validation in
+# customizations.yaml
+disable_xname_in_charts() {
+	yq w -i "${TMPDIR}/customizations.yaml" -- 'spec.kubernetes.services.cray-opa.opa.xnamePolicy.bos' 'false'
+	yq w -i "${TMPDIR}/customizations.yaml" -- 'spec.kubernetes.services.cray-opa.opa.xnamePolicy.cfs' 'false'
+	yq w -i "${TMPDIR}/customizations.yaml" -- 'spec.kubernetes.services.cray-opa.opa.xnamePolicy.ckdump' 'false'
+	yq w -i "${TMPDIR}/customizations.yaml" -- 'spec.kubernetes.services.cray-opa.opa.xnamePolicy.dvs' 'false'
+	yq w -i "${TMPDIR}/customizations.yaml" -- 'spec.kubernetes.services.cray-opa.opa.xnamePolicy.enabled' 'false'
+	yq w -i "${TMPDIR}/customizations.yaml" -- 'spec.kubernetes.services.spire.server.tokenService.enableXNameWorkloads' 'false'
+}
+
+# create_manifest creates the xname manifest with the cray-opa and spire information from
+# the manifests that ship with CSM. After this it will use manifestgen to create a manifest.yaml
+# to use with loftsman
+create_manifest() {
+	yq m -a append "${PWD}/manifests/sysmgmt.yaml" "${PWD}/manifests/platform.yaml" >"${TMPDIR}/xnamevalidation.yaml"
+	yq w -i "${TMPDIR}/xnamevalidation.yaml" "metadata.name" "xnamevalidation"
+
+	for chart in $(yq r "${TMPDIR}/xnamevalidation.yaml" 'spec.charts[*].name' | grep -Ev '(^cray-opa$|^spire$)'); do
+		yq d -i "${TMPDIR}/xnamevalidation.yaml" 'spec.charts(name=='"$chart"')'
+	done
+
+	manifestgen -c "${TMPDIR}/customizations.yaml" -i "${TMPDIR}/xnamevalidation.yaml" -o "${TMPDIR}/manifest.yaml"
+}
+
+# update_customizations saves the updated customizations file back to the
+# site-init secret
+update_customizations() {
+	CUSTOMIZATIONS="$(base64 <"${TMPDIR}/customizations.yaml" | tr -d '\n')"
+	kubectl get secrets -n loftsman site-init -o json \
+		| jq ".data.\"customizations.yaml\" |= \"$CUSTOMIZATIONS\"" | kubectl apply -f -
+}
+
+# run_loftsman runs loftsman against our trimmed down manifest file to enable
+# xname validation in the cray-opa and spire charts
+run_loftsman() {
+	loftsman ship --charts-path "${PWD}/helm" --manifest-path "${TMPDIR}/manifest.yaml"
+
+	# Restart opa pods so that the policy changes are picked up
+	kubectl rollout restart -n opa deployment cray-opa-ingressgateway
+	kubectl rollout restart -n opa deployment cray-opa-ingressgateway-customer-admin
+	kubectl rollout restart -n opa deployment cray-opa-ingressgateway-customer-user
+}
+
```

Comment:
I am not an expert. The exit 3s in validate_prereqs will cause the function to exit back to where the function is called by another function, but not the script itself.